diff --git a/dist/esm/chainHead/chainHead.mjs b/dist/esm/chainHead/chainHead.mjs
index 694f29035186d89e0db3d4171d05a939f026570e..bff0a740c625f3bfb56938d2b8859d00d4a53b52 100644
--- a/dist/esm/chainHead/chainHead.mjs
+++ b/dist/esm/chainHead/chainHead.mjs
@@ -157,7 +157,7 @@ const getChainHead$ = (chainHead) => {
     best$.pipe(map((b) => b.hash))
   );
   const _body$ = withOptionalHash$(commonEnhancer(lazyFollower("body")));
-  const body$ = (hash) => upsertCachedStream(hash, "body", _body$(hash, true));
+  const body$ = (hash, canonical) => upsertCachedStream(hash, "body", _body$(hash, canonical));
   const _storage$ = commonEnhancer(lazyFollower("storage"));
   const storage$ = withOptionalHash$(
     (hash, withCanonicalChain2, type, keyMapper, childTrie = null, mapper) => pinnedBlocks$.pipe(
@@ -244,11 +244,11 @@ const getChainHead$ = (chainHead) => {
       runtime$,
       metadata$,
       header$,
-      body$,
-      call$: withCanonicalChain(call$),
-      storage$: withCanonicalChain(storage$),
+      body$: withCanonicalChain(body$, false),
+      call$: withCanonicalChain(call$, false),
+      storage$: withCanonicalChain(storage$, false),
       storageQueries$,
-      eventsAt$: withCanonicalChain(eventsAt$),
+      eventsAt$: withCanonicalChain(eventsAt$, false),
       trackTx$,
       trackTxWithoutEvents$,
       validateTx$,
